src/main/java/org/ilgcc/jobs/CCMSSubmissionPayloadTransactionJob.java

```diff
@@ -55,6 +55,12 @@ public void sendCCMSTransaction(Submission submission) throws JsonProcessingExce
         CCMSTransaction ccmsTransaction = ccmsTransactionPayloadService.generateSubmissionTransactionPayload(submission);
         JsonNode response = ccmsApiClient.sendRequest(APP_SUBMISSION_ENDPOINT.getValue(), ccmsTransaction);
         log.info("Received response from CCMS when sending transaction payload: {}", response);
-        transactionRepositoryService.createTransaction(UUID.fromString(response.get("transactionId").asText()), submission.getId());
+
+        String workItemId = response.get("workItemId") != null ? response.get("workItemId").asText() : null;
```

Comment:
PR looks good!

Small suggestion here:

```suggestion
        String workItemId = response.hasNonNull("workItemId") ? response.get("workItemId").asText() : null;

if (workItemId == null) {
    log.warn("Received null work item ID from CCMS transaction for submission : {}", submission.getId());
}
```

Take it or leave it though, your way is fine too!